### ISSUE TITLE
Can't overfit one image

### ISSUE THREAD

**Issue by sdimantsd** [ultralytics/yolov5#8392]
### Search before asking

- [X] I have searched the YOLOv5 [issues](https://github.com/ultralytics/yolov5/issues) and [discussions](https://github.com/ultralytics/yolov5/discussions) and found no similar questions.

### Question

Hi, 
I am try to using you'r network. and unfortunately one of my trains not working...
I am try to recognize digits in license plate
This is the image with the labels:
![image](https://user-images.githubusercontent.com/29533264/176434270-0fc4df2a-8435-4feb-b721-98b8ade20eeb.png)

If I train with yolov4 it's works perfect. but in yolov5 it not recognize a thing.
I tried to remove all labels except the '5' in the .txt file but it still not working.
I tried many images sizes, different architectures (yolov5s, yolov5m) and still nothing.
Can you please help me?

The YAML file:
```

# Train/val/test sets as 1) dir: path/to/imgs, 2) file: path/to/imgs.txt, or 3) list: [path/to/imgs1, path/to/imgs2, ..]
path: /home/ws/data/dataset/digits_overfitt/  # dataset root dir
train: overfitt.txt  # train images (relative to 'path') 128 images
val: overfitt.txt  # val images (relative to 'path') 128 images
test:  # test images (optional)

# Classes
nc: 10  # number of classes
names: ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9']  # class names

```
  
The  "overfitt.txt" file:
```
7 0.15680473372781065 0.8017751479289941 0.20710059171597633 0.21893491124260356
8 0.26331360946745563 0.7041420118343196 0.20710059171597633 0.21301775147928995
8 0.4171597633136095 0.5769230769230769 0.1952662721893491 0.1834319526627219
8 0.514792899408284 0.4911242603550296 0.20118343195266272 0.20118343195266272
5 0.6035502958579881 0.40828402366863903 0.21301775147928995 0.21301775147928995
7 0.7485207100591716 0.2958579881656805 0.20710059171597633 0.22485207100591717
8 0.8431952662721893 0.21301775147928995 0.20710059171597633 0.21301775147928995
```

Thanks

**Comment by glenn-jocher** [comment]
👋 Hello! Thanks for asking about improving YOLOv5 🚀 training results. 

Most of the time good results can be obtained with no changes to the models or training settings, **provided your dataset is sufficiently large and well labelled**. If at first you don't get good results, there are steps you might be able to take to improve, but we always recommend users **first train with all default settings** before considering any changes. This helps establish a performance baseline and spot areas for improvement.

If you have questions about your training results **we recommend you provide the maximum amount of information possible** if you expect a helpful response, including results plots (train losses, val losses, P, R, mAP), PR curve, confusion matrix, training mosaics, test results and dataset statistics images such as labels.png. All of these are located in your `project/name` directory, typically `yolov5/runs/train/exp`.

We've put together a full guide for users looking to get the best results on their YOLOv5 trainings below.

## Dataset

- **Images per class.** ≥ 1500 images per class recommended
- **Instances per class.** ≥ 10000 instances (labeled objects) per class recommended
- **Image variety.** Must be representative of deployed environment. For real-world use cases we recommend images from different times of day, different seasons, different weather, different lighting, different angles, different sources (scraped online, collected locally, different cameras) etc.
- **Label consistency.** All instances of all classes in all images must be labelled. Partial labelling will not work.
- **Label accuracy.** Labels must closely enclose each object. No space should exist between an object and it's bounding box. No objects should be missing a label.
- **Label verification.** View `train_batch*.jpg` on train start to verify your labels appear correct, i.e. see [example](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data#local-logging) mosaic.
- **Background images.** Background images are images with no objects that are added to a dataset to reduce False Positives (FP). We recommend about 0-10% background images to help reduce FPs (COCO has 1000 background images for reference, 1% of the total). No labels are required for background images.

<a href="https://arxiv.org/abs/1405.0312"><img width="800" src="https://user-images.githubusercontent.com/26833433/109398377-82b0ac00-78f1-11eb-9c76-cc7820669d0d.png" alt="COCO Analysis"></a>


## Model Selection

Larger models like YOLOv5x and [YOLOv5x6](https://github.com/ultralytics/yolov5/releases/tag/v5.0) will produce better results in nearly all cases, but have more parameters, require more CUDA memory to train, and are slower to run. For **mobile** deployments we recommend YOLOv5s/m, for **cloud** deployments we recommend YOLOv5l/x. See our README [table](https://github.com/ultralytics/yolov5#pretrained-checkpoints) for a full comparison of all models. 

<p align="center"><img width="700" alt="YOLOv5 Models" src="https://github.com/ultralytics/yolov5/releases/download/v1.0/model_comparison.png"></p>

- **Start from Pretrained weights.** Recommended for small to medium sized datasets (i.e. [VOC](https://github.com/ultralytics/yolov5/blob/master/data/VOC.yaml), [VisDrone](https://github.com/ultralytics/yolov5/blob/master/data/VisDrone.yaml), [GlobalWheat](https://github.com/ultralytics/yolov5/blob/master/data/GlobalWheat2020.yaml)). Pass the name of the model to the `--weights` argument. Models download automatically from the [latest YOLOv5 release](https://github.com/ultralytics/yolov5/releases).
```bash
python train.py --data custom.yaml --weights yolov5s.pt
                                             yolov5m.pt
                                             yolov5l.pt
                                             yolov5x.pt
                                             custom_pretrained.pt
```
- **Start from Scratch.** Recommended for large datasets (i.e. [COCO](https://github.com/ultralytics/yolov5/blob/master/data/coco.yaml), [Objects365](https://github.com/ultralytics/yolov5/blob/master/data/Objects365.yaml), [OIv6](https://storage.googleapis.com/openimages/web/index.html)). Pass the model architecture yaml you are interested in, along with an empty `--weights ''` argument:
```bash
python train.py --data custom.yaml --weights '' --cfg yolov5s.yaml
                                                      yolov5m.yaml
                                                      yolov5l.yaml
                                                      yolov5x.yaml
```


## Training Settings

Before modifying anything, **first train with default settings to establish a performance baseline**. A full list of train.py settings can be found in the [train.py](https://github.com/ultralytics/yolov5/blob/master/train.py) argparser.

- **Epochs.** Start with 300 epochs. If this overfits early then you can reduce epochs. If overfitting does not occur after 300 epochs, train longer, i.e. 600, 1200 etc epochs.
- **Image size.** COCO trains at native resolution of `--img 640`, though due to the high amount of small objects in the dataset it can benefit from training at higher resolutions such as `--img 1280`. If there are many small objects then custom datasets will benefit from training at native or higher resolution. Best inference results are obtained at the same `--img` as the training was run at, i.e. if you train at `--img 1280` you should also test and detect at `--img 1280`.
- **Batch size.** Use the largest `--batch-size` that your hardware allows for. Small batch sizes produce poor batchnorm statistics and should be avoided.
- **Hyperparameters.** Default hyperparameters are in [hyp.scratch-low.yaml](https://github.com/ultralytics/yolov5/blob/master/data/hyps/hyp.scratch-low.yaml). We recommend you train with default hyperparameters first before thinking of modifying any. In general, increasing augmentation hyperparameters will reduce and delay overfitting, allowing for longer trainings and higher final mAP. Reduction in loss component gain hyperparameters like `hyp['obj']` will help reduce overfitting in those specific loss components. For an automated method of optimizing these hyperparameters, see our [Hyperparameter Evolution Tutorial](https://docs.ultralytics.com/yolov5/tutorials/hyperparameter_evolution).

## Further Reading

If you'd like to know more a good place to start is Karpathy's 'Recipe for Training Neural Networks', which has great ideas for training that apply broadly across all ML domains: http://karpathy.github.io/2019/04/25/recipe/

Good luck 🍀 and let us know if you have any other questions!

**Comment by sdimantsd** [comment]
Hi, 
Thanks for the answer.
I am using the default setting and I manage to train different datasets (coco128 and my own dataset). 
But the training process is not working for this dataset (the digits dataset, even for overfit).

Can you help me please?
Thanks

**Comment by barney2074** [comment]
probably not the main problem, but your labels don't seem very well fitted to the objects i.e too much background

![image](https://user-images.githubusercontent.com/25944617/176564190-50ae5c4b-f454-4fba-a806-f4d7101a6afd.png)

**Comment by sdimantsd** [comment]
You right. 
But it's work on yolov4.
And also, it not overfitting even one image.

**Comment by barney2074** [comment]
@sdimantsd 

I don't think your YAML is correct
There are options 1-3 to either point at a directory, or a list of images
So your txt file should contain a list of images, not the label data
https://docs.ultralytics.com/tutorials/train-custom-datasets/
The Roboflow has a useful summary of various formats https://roboflow.com/formats

Also- 'overfitting' means the model fits the training data too well, but not unseen images

![image](https://user-images.githubusercontent.com/25944617/176799257-440d3591-1899-4765-8f13-106309d6eda8.png)


Andrew

**Comment by sdimantsd** [comment]
Sorry, 
The overfitt.txt contain one row with the image name.
The image label file is the file with the labels. my mistake.
Anyway, I don't know what was the problem. but it's works now.
Thank!

**Comment by glenn-jocher** [comment]
@sdimantsd glad to hear it works now! If you encounter any further issues or have any questions, don't hesitate to ask. 😊 Keep up the great work!
